mlkem/native/x86_64/src/basemul.S

```diff
@@ -83,21 +87,6 @@ vpsubw		%ymm11,%ymm3,%ymm11			# a1d1
 vpsubw		%ymm7,%ymm0,%ymm7			# b1c1
 vpsubw		%ymm12,%ymm4,%ymm12			# b1d1
 
-/* b0*d0 and b1*d1 need twisting by a twiddle, accounting
- * for X^2=zeta in F_q[X]/(X^2-zeta).
- *
- * TODO: This could be precomputed in the mulcache */
-vmovdqa		(%r9),%ymm0
-vmovdqa		32(%r9),%ymm1
-vpmullw		%ymm0,%ymm10,%ymm2
-vpmullw		%ymm0,%ymm12,%ymm3
-vpmulhw		%ymm1,%ymm10,%ymm10
-vpmulhw		%ymm1,%ymm12,%ymm12
-vpmulhw		%ymm8,%ymm2,%ymm2
-vpmulhw		%ymm8,%ymm3,%ymm3
-vpsubw		%ymm2,%ymm10,%ymm10			# rb0d0
-vpsubw		%ymm3,%ymm12,%ymm12			# rb1d1
-
 vpaddw		%ymm5,%ymm9,%ymm9
 vpaddw		%ymm7,%ymm11,%ymm11
 vpsubw		%ymm13,%ymm10,%ymm13
```

Comment:
Pre-existing, but let's improve while we're at it: It has to be noted that the order of arguments in the last two subs is different, removing the sign from `-b1d1z`. Hence we get `a1b1 + b1d1z` as desired.

Reply:
Fixed this